test/test_checker.ts

```diff
@@ -239,6 +241,21 @@ describe("ts-interface-checker", () => {
       /value.kind is not a valid enum value/);
   });
 
+  it("should handle intersections", () => {
+    const { Car, House } = createCheckers(intersectionTI);
+    Car.check({ numDoors: 2, numWheels: 4})
+    House.check({ numDoors: 6, numRooms: 8 })
+
+    assert.throws(() => Car.check({ numDoors: 2 }),
+      /value.numWheels is missing/);
+    assert.throws(() => Car.check({ numWheels: 4 }),
+      /value.numDoors is missing/);
+    assert.throws(() => House.check({ numDoors: 6 }),
+      /value.numRooms is missing/);
+    assert.throws(() => House.check({ numRooms: 8 }),
```

Comment:
I think some variety of tests would be useful here: a type with more than 2 parts of intersection, a check that fails more than one of them, a check that fails in a different way (e.g. `{numDoors: "foo"}`. Also a strictCheck() (but see earlier comment).

Reply:
Okay, i believe I was able to implement strict check correctly, as well as added some more test cases from your suggestions.